Library/Homebrew/utils/analytics.rb

```diff
@@ -1,34 +1,52 @@
-
-def analytics_anonymous_prefix_and_os
+def analytics_label
   @analytics_anonymous_prefix_and_os ||= begin
-    "#{OS_VERSION}, #{HOMEBREW_PREFIX.to_s.gsub(ENV["HOME"], "~")}"
+    os = OS_VERSION
+    prefix = ", non-/usr/local" if HOMEBREW_PREFIX.to_s != "/usr/local"
+    ci = ", CI" if ENV["CI"]
+    "#{os}#{prefix}#{ci}"
   end
 end
 
 def report_analytics(type, metadata={})
   return unless ENV["HOMEBREW_ANALYTICS"]
+  return if ENV["HOMEBREW_NO_ANALYTICS"]
 
-  metadata_args = metadata.map do |key, value|
-    ["-d", "#{key}=#{value}"] if key && value
-  end.compact.flatten
+  debug = !!ENV["HOMEBREW_ANALYTICS_DEBUG"]
 
-  # Send analytics. Anonymise the IP address (aip=1) and don't send or store
-  # any personally identifiable information.
-  # https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide
-  # https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
-  system ENV["HOMEBREW_CURL"], "https://www.google-analytics.com/collect",
-    "-d", "v=1", "--silent", "--max-time", "3", "--output", "/dev/null",
+  args = [
+    "--max-time", "3",
     "--user-agent", HOMEBREW_USER_AGENT_CURL,
+    "-d", "v=1",
     "-d", "tid=#{ENV["HOMEBREW_ANALYTICS_ID"]}",
     "-d", "cid=#{ENV["HOMEBREW_ANALYTICS_USER_UUID"]}",
     "-d", "aip=1",
-    "-d", "an=Homebrew",
+    "-d", "an=#{HOMEBREW_PRODUCT}",
     "-d", "av=#{HOMEBREW_VERSION}",
     "-d", "t=#{type}",
-    *metadata_args
+  ]
+  metadata.each {|k, v| args << "-d" << "#{k}=#{v}" if k && v }
+
+  if debug
+    puts Utils.popen_read ENV["HOMEBREW_CURL"],
+      "https://www.google-analytics.com/debug/collect",
+      *args
+  else
+    # Send analytics. Anonymise the IP address (aip=1) and don't send or store
+    # any personally identifiable information.
```

Comment:
This looks like it could/should be placed somewhere else, as `aip=1` is set up a bit further up.